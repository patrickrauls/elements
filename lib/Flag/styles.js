import { css } from 'styled-components';

const styles = css`
  &.invalid-flag,
  &.required-flag {
    position: absolute;
    right: -0.75rem;
    bottom: 0.25rem;
    font-size: 0.875rem;
    font-weight: 600;
    color: white;
    z-index: 1;
    background: linear-gradient(109deg, #af1a12 0px, #af1a12 30%, #eb483e 100%);
    text-align: center;
    line-height: 1.5rem;
    height: 1.5625rem;
    border-radius: 3px;
    width: 5.75rem;
    &:before {
      position: absolute;
      left: -0.3rem;
      top: 0.3rem;
      content: '';
      width: 0;
      height: 0;
      border-top: 7px solid transparent;
      border-right: 7px solid #af1a12;
      border-bottom: 7px solid transparent;
    }
  }
  &.ellies-input-valid-indicator {
    position: absolute;
    right: 0;
    bottom: 0.125rem;
    animation: fadein 0.3s;
  }
`;

export default styles;
