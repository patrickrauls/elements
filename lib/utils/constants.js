export const FAILED_VALIDATION =
  'Validation requirements have not been satisfied.';

export const regEx = {
  email: /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/
}

export const emails = {
  valid: [
    {
      value: 'email@domain.com',
      description: 'email is standard'
    },
    {
      value: 'firstname.lastname@domain.com',
      description: 'email contains dot in the address field'
    },
    {
      value: 'email@subdomain.domain.com',
      description: 'Email contains dot with subdomain'
    },
    {
      value: 'firstname+lastname@domain.com',
      description: 'Plus sign is considered valid character'
    },
    {
      value: 'email@123.123.123.123',
      description: 'Domain is valid IP address'
    },
    {
      value: 'email@[123.123.123.123]',
      description: 'Square bracket around IP address is considered valid'
    },
    {
      value: '"email"@domain.com',
      description: 'Quotes around email is considered valid'
    },
    {
      value: '1234567890@domain.com',
      description: 'Digits in address are valid'
    }, {
      value: 'email@domain-one.com',
      description: 'Dash in domain name is valid'
    }, {
      value: '_______@domain.com',
      description: 'Underscore in the address field is valid'
    }, {
      value: 'email@domain.name',
      description: '.name is valid Top Level Domain name'
    }, {
      value: 'email@domain.co.uk',
      description: 'Dot in Top Level Domain name also considered valid'
    }, {
      value: 'firstname-lastname@domain.com',
      description: 'Dash in address field is valid'
    }
  ],
  invalid: [
    {
      value: 'plainaddress',
      description: 'Missing @sign and domain'
    }, {
      value: '#@%^%#$@#$@#.com',
      description: 'Garbage'
    }, {
      value: '@domain.com',
      description: 'Missing username'
    }, {
      value: 'email.domain.com',
      description: 'Missing @'
    }, {
      value: 'email.@domain.com',
      description: 'Trailing dot in address is not allowed'
    }, {
      value: 'あいうえお@domain.com',
      description: 'Unicode char as address'
    }, {
      value: 'email@domain',
      description: 'Missing top level domain'
    }, {
      value: 'email@-domain.com',
      description: 'Leading dash in front of domain is invalid'
    }, {
      value: 'email@domain..com',
      description: 'Multiple consecutive dots in the domain portion is invalid'
    }
  ]
};
