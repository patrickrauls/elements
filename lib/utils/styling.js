import { keyframes } from 'styled-components';

export const icons = {
  submit: 'fa fa-chevron-right',
  link: 'fa fa-chevron-right',
  loading: 'fas fa-circle-notch fast-spin',
  big: 'fa-2x',
  valid: 'fa fa-check'
};

export const fastSpin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(720deg);
  }
`;