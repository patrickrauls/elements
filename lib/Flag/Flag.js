import React from 'react';
import PropTypes from 'prop-types';
import { icons } from '../utils/styling';

const Flag = ({ className, type }) => {
  switch (type) {
    case 'hidden':
      return <span className="hidden" />;
    case 'valid':
      return (
        <i
          className={`
            ${className}
            valid-indicator
            ${icons.valid}
          `}
        />
      );
    default:
      return (
        <span
          className={`
            ${className}
            ${type}-flag
            flag
          `}
        >
          {type}
        </span>
      );
  }
};

Flag.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf(['valid', 'required', 'invalid', 'hidden']).isRequired
};

Flag.defaultProps = {
  className: ''
};

export default Flag;
