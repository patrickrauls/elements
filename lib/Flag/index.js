import styled from 'styled-components';
import Flag from './Flag';
import styles from './styles';

const StyledFlag = styled(Flag)`
  ${styles};
`;

export default StyledFlag;
