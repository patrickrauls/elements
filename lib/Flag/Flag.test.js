import React from 'react';

import Flag from './Flag';
import StyledFlag from './index';
import { icons } from '../utils/styling';

describe('Flag', () => {
  describe('unit tests', () => {
    let wrapper;
    const setup = props => shallow(<Flag type="hidden" {...props} />);
    it('should render without crashing', () => {
      wrapper = setup();
      expect(wrapper.exists()).toBe(true);
    });
    it('should add .hidden when given a type of hidden', () => {
      wrapper = setup({ type: "hidden" });
      expect(wrapper.hasClass('hidden')).toBe(true);
    });
    it('should show a valid icon when given a type of valid', () => {
      wrapper = setup({ type: 'valid' });
      expect(wrapper.hasClass(icons.valid)).toBe(true);
    });
    it('should show text of invalid when given an type of invalid', () => {
      wrapper = setup({ type: 'invalid' });
      expect(wrapper.text()).toEqual('invalid');
    });
    it('should show text of required when given an type of required', () => {
      wrapper = setup({ type: 'required' });
      expect(wrapper.text()).toEqual('required');
    });
  });
  describe('ui tests', () => {
    const setup = props => shallow(<StyledFlag type="valid" {...props} />);
    it('should add styles to the component', () => {
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
