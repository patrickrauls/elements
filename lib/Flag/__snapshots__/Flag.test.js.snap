// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag ui tests should add styles to the component 1`] = `
<Flag
  className="sc-bdVaJa hrWjhA"
  type="valid"
/>
`;
